### DISCUSSION TITLE
ComfyUI for Intel Evo

### DISCUSSION THREAD

**Discussion by aaaaa-a-aa** [Comfy-Org/ComfyUI#1942] [Q&A] [answered]
Is there any working going on for making ComfyUI suitable for "Intel Evo"? Since Colab banned free usage for stable diffusion, its annoying. Making it available for Intel Evo would be highly appreciated.

**Comment by aaaaa-a-aa** [comment]
@kwaa Just asking cause you were in Intel Arc topic. Is this a topic on work?

**Reply by kwaa** [reply]
It's really up to Intel, and downstream doesn't play much of a role in that.

https://intel.github.io/intel-extension-for-pytorch/xpu/latest/tutorials/installations/linux.html#hardware-requirement

https://github.com/intel/intel-extension-for-pytorch/issues/292
https://github.com/intel/intel-extension-for-pytorch/issues/339
https://github.com/intel/intel-extension-for-pytorch/issues/406

**Reply by simonlui** [reply, marked as answer]
Supporting Intel iGPU would probably require implementing OpenVINO in order for "Intel Evo" branded laptops to work. IPEX only works officially for Arc at the moment, but given the graphics architecture, it can also include Tiger Lake and up but not officially. Intel Evo can be anything from Comet Lake to Alder Lake mobile so it doesn't mean anything and that small sliver from Comet Lake to Ice Lake can mean a good portion of some random Intel Evo branded laptop would not be supported.
I did some private work playing around with implementing it but the main issue is redundancy of functionality where it won't help out drastically that much with iGPUs and gives another backend for Intel GPUs for no real good reason vs just using CPU and the speeds which you can use this even with OpenVINO and any old SD model without quantizing the models with iGPUs is atrociously slow. I haven't asked but due to that, I surmised that there would be a high possibility comfyanonymous would reject such a patch and thus didn't really pursue it further. Maybe Meteor Lake and the NPU included can change that but I don't see a future for implementing it.
